spec/killbill_client/encoding_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'spec_helper'
+
+describe KillBillClient::API do
+  let(:expected_query_params) {
```

Comment:
This is still not deterministic: prior to 1.9, behavior of enumerated hashes was not in the ruby specification and therefore was up to implementation (in 1.9+, hash enumeration is specified by the language to be in the order of insertion). On Ruby 1.8, the output of `encode_params` is not predictable.

How about decoding `query_string` below?

```
url = 'http://www.foo.com?pluginProperty=details%3D%257B%2522eventType%2522%253A%2522voidEvent%2522%252C%2522transactionType%2522%253A%2522void%2522%257D'
puts CGI.parse(URI.parse(url).query)
{"pluginProperty"=>["details=%7B%22eventType%22%3A%22voidEvent%22%2C%22transactionType%22%3A%22void%22%7D"]}
```